python/src/spark_rapids_ml/umap.py

```diff
@@ -1141,19 +1179,26 @@ def _train_udf(pdf_iter: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
             # *note*: cuml_fit_func may delete components of inputs to free
             # memory.  do not rely on inputs after this call.
             embedding, raw_data = cuml_fit_func(inputs, params).values()
+
             logger.info("Cuml fit complete")
 
             num_sections = (len(embedding) + chunk_size - 1) // chunk_size
 
             for i in range(num_sections):
                 start = i * chunk_size
                 end = min((i + 1) * chunk_size, len(embedding))
+                if self._sparse_fit:
+                    # return sparse array as a list in COO format
+                    coo = raw_data[start:end].tocoo()
+                    raw_data = list(zip(coo.row, coo.col, coo.data))
```

Comment:
Good point. One option would be to return a list of dicts so that we can use a StructField and specify a separate type for each, but at the cost of memory overhead. 
Or we can return the row, col, and data lists as separate arrays and reconcatenate into COO LIL on the driver.

Reply:
yeah, the idea sounds to work. Is it possible that we change the out_schema data type (whether float or int) depending on whether the cuml fit runs on dense or sparse matrix? out_schema is invoked after fit so we may be able to dynamically determine the output schema.